be/src/formats/parquet/file_reader.cpp

```diff
@@ -189,9 +217,40 @@ StatusOr<bool> FileReader::_filter_group(const tparquet::RowGroup& row_group) {
 Status FileReader::_read_min_max_chunk(const tparquet::RowGroup& row_group, const std::vector<SlotDescriptor*>& slots,
                                        ChunkPtr* min_chunk, ChunkPtr* max_chunk, bool* exist) const {
     const HdfsScannerContext& ctx = *_scanner_ctx;
+
+    // Key is column name, the same as name in SlotDescription, always case sensitive
+    // This map only used to get ColumnMeta through Iceberg Schema
+    std::unordered_map<std::string, size_t> column_name_2_pos_in_meta{};
+    if (_scanner_ctx->iceberg_schema != nullptr) {
+        std::unordered_map<std::string, int32_t> column_name_2_field_id{};
+        for (auto each : _scanner_ctx->iceberg_schema->fields) {
+            const std::string& field_name =
+                    _scanner_ctx->case_sensitive ? each.name : boost::algorithm::to_lower_copy(each.name);
+            column_name_2_field_id.emplace(field_name, each.field_id);
+        }
+
+        for (const auto& slot : slots) {
+            const std::string& need_column_name =
+                    _scanner_ctx->case_sensitive ? slot->col_name() : boost::algorithm::to_lower_copy(slot->col_name());
+            auto it = column_name_2_field_id.find(need_column_name);
+            if (it == column_name_2_field_id.end()) {
+                continue;
+            }
+            // Put SlotDescriptor's column name here
+            column_name_2_pos_in_meta.emplace(slot->col_name(),
+                                              _file_metadata->schema().get_field_pos_by_field_id(it->second));
+        }
+    }
+
     for (size_t i = 0; i < slots.size(); i++) {
         const SlotDescriptor* slot = slots[i];
-        const auto* column_meta = _get_column_meta(row_group, slot->col_name(), _scanner_ctx->case_sensitive);
+        const tparquet::ColumnMetaData* column_meta = nullptr;
+        if (_scanner_ctx->iceberg_schema == nullptr) {
+            column_meta =
```

Comment:
there are a lot of if/else here? can we extract interface and gives them two implementations?

Reply:
improved